cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -427,6 +455,21 @@ func (c *Client) AddMember(name string, peerAddrs string) ([]Member, error) {
 	return ret, nil
 }
 
+func memberPromoteWithRety(ctx context.Context, cli *clientv3.Client, learnerID uint64) error {
+	klog.V(1).Infof("[etcd] Promoting a learner as a voting member: %016x", learnerID)
+	err := wait.PollImmediate(15*time.Second, 10*time.Minute, func() (bool, error) {
+		// TODO: warning logs from etcd client should be removed
+		_, err := cli.MemberPromote(ctx, learnerID)
+		if err == nil {
+			klog.V(1).Infof("[etcd] The learner was promoted as a voting member: %016x", learnerID)
+			return true, nil
+		}
+		klog.V(4).Infof("[etcd] Promoting the learner %016x failed: %v", learnerID, err)
```

Comment:
I think that any `err` should have a higher priority to log, unless it intends to consistent with others.

Reply:
1. as the TODO described, the etcd client will print warnings as well. This log will be duplicated if we print it as an error here.
2. there will be retrying and etcd member promote should wait for a period until `data synced`. The period depends on the data size of etcd and network & disk. 
3. the error will be returned if all retries failed, and the error message will be printed in the caller.

Reply:
```
		// TODO: warning logs from etcd client should be removed.
		// The warning logs are printed by etcd client code for several reasons, including
		// 1. can not promote yet(no synced)
		// 2. context deadline exceeded
		// 3. peer URLs already exists
		// Once the client provides a way to check if the etcd learner is ready to promote, the retry logic can be replaced.
```
I tried to summarize why this is a TODO.

Reply:
> This log will be duplicated if we print it as an error here. 

I meant you can print log with `klog.V(1)`, the reason is that you print some info with `klog.V(1)` while an error with level 4, that doesn't sound reasonable.

Reply:
I still cannot persuade myself that we should `V(1)` log a retry error message that will finally be gone.

Reply:
make sense!